tests/testImageMapReduce.py

```diff
@@ -0,0 +1,243 @@
+        self.assertClose(newArr.max(), 5.)
+
+    def testExampleTaskWithOverlaps(self):
+        """Test sample grid task that adds 5.0 to input image and uses
+        'average' `reduceOperation`.
+        """
+        config = TestImageMapReduceConfig()
+        config.gridStepX = config.gridStepY = 8
+        config.reducerSubtask.reduceOperation = 'average'
+        task = ImageMapReduceTask(config)
+        config.mapperSubtask.addAmount = 5.
+        newExp = task.run(self.exposure)
+        newMI = newExp.getMaskedImage()
+        newArr = newMI.getImage().getArray()
+        mi = self.exposure.getMaskedImage()
+
+        self.assertClose(mi.getImage().getArray().mean(), newArr.mean() - 5.)
+        self.assertClose(newArr.mean(), 5.)
+        self.assertClose(newArr.min(), 5.)
+        self.assertClose(newArr.max(), 5.)
+
+    def testExampleTaskMean(self):
+        """Test sample grid task that returns the mean of the subimages and uses
+        'none' `reduceOperation`.
+        """
+        config = TestImageMapReduceConfig2()
+        config.gridStepX = config.gridStepY = 8
+        config.reducerSubtask.reduceOperation = 'none'
+        task = ImageMapReduceTask(config)
+        subMeans = task.run(self.exposure)
+
+        self.assertEqual(len(subMeans), len(task.boxes0))
+        self.assertClose(subMeans[0], 0.)
+
+    def testGridCentroids(self):
+        """Test sample grid task which is provided a set of `gridCentroids` and
+        returns the mean of the subimages surrounding those centroids using 'none'
+        for `reduceOperation`.
+        """
+        config = TestImageMapReduceConfig2()
+        config.gridStepX = config.gridStepY = 8
+        config.reducerSubtask.reduceOperation = 'none'
+        centroidsX = [i for i in np.linspace(0, 128, 50)]
+        centroidsY = centroidsX
+        config.gridCentroidsX = centroidsX
+        config.gridCentroidsY = centroidsY
+        task = ImageMapReduceTask(config)
+        subMeans = task.run(self.exposure)
+
+        self.assertEqual(len(subMeans), len(centroidsX))
+        self.assertClose(subMeans[0], 0.)
+
+    def testGridValidity(self):
+        """!Test sample grids with various spacings and sizes and other options.
+
+        Try to see if we can break it.
+        """
+        config = TestImageMapReduceConfig()
+        config.reducerSubtask.reduceOperation = 'copy'
+        expectedVal = 1.
+        config.mapperSubtask.addAmount = expectedVal
+
+        for scaleByFwhm in (False, True):
+            config.scaleByFwhm = scaleByFwhm
+            for gsx in range(51, 1, -10):
+                config.gridStepX = config.gridSizeX = gsx
+                for gsy in range(51, 1, -10):
+                    config.gridStepY = config.gridSizeY = gsy
+
+                    task = ImageMapReduceTask(config)
+                    boxes = task._generateGrid(self.exposure)
+                    if len(boxes[0]) > 1000:  # bypass to prevent slow testing
```

Comment:
I'm uncomfortable with a "just in case" thing that disables tests. If you've tweaked the parameters so that it doesn't happen, just get rid of it. If a future person re-changes the parameters such that the test takes a long time, they can figure out what to do then.